### ISSUE TITLE
Migrate database to Trove

### ISSUE THREAD

**Issue by audiodude** [openzim/wp1#731]
With the future addition of "standard" selections (aka those on download.openzim.org) to WP1, we will need to provision a much larger database. Current database is ~32 GB but we estimate that we will need ~ 75 GB in the future. This requires a database migration off of our current Toolforge db to a Wikimedia Cloud Trove instance. See https://phabricator.wikimedia.org/T363901.

I anticipate that the most straightforward way to do this is:

1. Turn off the WP1 server and workers, preferably at a time when the WP 1.0 Bot is not running.
2. Create a mysql dump of the current database
3. Restore the dump to Trove
4. Update the production config with the new database credentials
5. Turn the WP1 server and workers back on

**Comment by audiodude** [comment]
It might be possible to restore directly from one of our `kiwix/borg-backup`s. However depending on when the backup was taken, it might result in some loss of data (especially user data).

Alternately, perhaps there is a way to "force backup" on `kiwix/borg-backup`, instead of doing a manual mysqldump, and restore from there. This has the advantage that the particular snapshot we use to restore will be stored and can be inspected later for potential sources of issues.

**Comment by audiodude** [comment]
@benoit74 @rgaudin @kelson42 WDYT?

**Comment by kelson42** [comment]
@audiodude LGTM, please backup system on third party storage before running scenario and keep old DB a few weeks before asking for removal.

**Comment by audiodude** [comment]
@kelson42 Can you recommend a service? Or is just a random spot on S3 or wasabi okay?

**Comment by kelson42** [comment]
@audiodude You can temporarly use one of the mwoffliner machines.

**Comment by audiodude** [comment]
Okay I have switched off the WP1 server and started the database backup/dump.

I have also created the new database and user in our Trove instance and updated the WP1 config. I have tested connectivity from the command line.

Presumably I will be able to restore from this file and turn WP1 back on and test it is working an no data was lost.

**Comment by audiodude** [comment]
There were IO problems with the Cloud VPS disk. I've changed strategies to do `mysqldump <db> | mysql -h <trove> <newdb>`. This will take about 2 hours, and I'll shut down the WP1 server first. I'll update here when that process begins.

**Comment by audiodude** [comment]
The throughput of the operation dropped to an unacceptable level of 1.5Mb/sec. I stopped it and restarted the server. Looking at other options currently.

**Comment by audiodude** [comment]
Took down WP1 for about 8 hours today and migrated the database with a direct pipe similar to `mysqldump <current db> | mysql <new db>`. There is no database dump file that was created. I've tested that wp1.openzim.org is working, with project data available, and I've created a new selection which I verified showed up in the new database. I've also reverted temporary unsafe configuration changes (around innodb transactions) that were applied to attempt to speed up the migration.

Overall average throughput ended up being 726Kb/sec 😭
